packages/client/src/suiEncode.ts

```diff
@@ -0,0 +1,49 @@
+import { ethers } from "ethers";
+import { z } from "zod";
+
+import { stringArraySchema } from "../../../types/shared.schema";
+import { validateAndParseSchema } from "../../../utils/validateAndParseSchema";
+
+const encodeOptionsSchema = z.object({
+  data: z.string({
+    invalid_type_error: "Data must be a string",
+    required_error: "Data is required",
+  }),
+  objects: stringArraySchema.optional(),
+  typeArguments: stringArraySchema.optional(),
+});
+
+export type EncodeOptions = z.infer<typeof encodeOptionsSchema>;
+
+/**
+ * Encodes data for Sui blockchain transactions using ethers.js
+ *
+ * @param {EncodeOptions} options - The encoding options
+ * @param {string} options.data - The data to encode
+ * @param {string[]} [options.typeArguments=[]] - Type arguments for generic functions
+ * @param {string[]} [options.objects=[]] - Object references to include
+ * @returns {string} The ABI-encoded data suitable for Sui transactions
+ */
+export const suiEncode = (options: EncodeOptions): string => {
```

Comment:
Does this really belong to the client folder? It's not really being used in the `ZetaChainClient` class, I see this as more of a standalone function that could go in utils. No big deal though.

Reply:
Yes, we'll need to refactor this. Perhaps, we need to separate util functions we extracted for reusability from the ones we want users to import. Mixing them in the same `utils` dir is not ideal, I think. We can refactor this in follow up PRs.